jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SniX509ExtendedKeyManager.java

```diff
@@ -81,29 +88,51 @@ protected String chooseServerAlias(String keyType, Principal[] issuers, Collecti
                 {
                     SslContextFactory.AliasSNIMatcher matcher = (SslContextFactory.AliasSNIMatcher)m;
                     host = matcher.getHost();
-                    x509 = matcher.getX509();
                     break;
                 }
             }
         }
 
-        if (LOG.isDebugEnabled())
-            LOG.debug("Matched {} with {} from {}", host, x509, Arrays.asList(aliases));
+        SslContextFactory.SNISelector sniSelector = _sslContextFactory.getSNISelector();
+        if (sniSelector == null)
```

Comment:
The _sslContextFactory should just return this rather than null.   Null should mean always reject.